resdk/tests/functional/duplication/e2e_duplication.py

```diff
@@ -0,0 +1,39 @@
+# pylint: disable=missing-docstring
+from resdk.exceptions import ResolweServerError
+from resdk.tests.functional.base import BaseResdkFunctionalTest
+
+
+class TestDuplication(BaseResdkFunctionalTest):
+
+    def test_collection_duplication(self):
+        collection = self.res.collection.create(name='Test collection')
+        duplicate = collection.duplicate()
+        self.assertEqual(duplicate.name, 'Copy of Test collection')
+
+    def test_sample_duplication(self):
+        sample = self.res.sample.create(name='Test sample')
+        duplicate = sample.duplicate()
+        self.assertEqual(duplicate.name, 'Copy of Test sample')
+
+    def test_sample_duplication_perms(self):
+        collection = self.user_res.collection.create()
+        sample = self.user_res.sample.create(name='Test sample')
+        collection.add_samples(sample)
+
+        collection.permissions.remove_user(self.user_res.auth.username, 'add')
```

Comment:
Use `USER_USERNAME` (form `resdk.tests.functional.base`) instead of `self.user_res.auth.username`

Reply:
Sidenote for @JureZmrzlikar: we should add constants with permission values, e.g. `view`, `edit`, `add`,...

Reply:
@hadalin it would be cleaner approach to create objects with `self.res` (admin user) and then grant permissions to `USER_USERNAME`.

Reply:
Regarding constants for view, edit, add, ... I'd rather spend effort polishing this at later time -when permissions changes take place.

Reply:
>  at later time

That's why it is a sidenote :)

Reply:
>it would be cleaner approach to create objects with self.res

Isn't it required to create a sample like this `sample = self.user_res.sample.create` so that `sample.duplicate()` makes a request as `'user'`?

Also isn't it more obvious what we're trying to test if we _remove_ the permission? If not, then please provide a code snippet, so I can see what you mean.